### ISSUE TITLE
Deploy key is not being used

### ISSUE THREAD

**Issue by vkurup** [caktus/tequila-django#20]
I'm not sure why this is not happening to anyone else. The first time through the web playbook, the git checkout works normally. But any subsequent time (either during the same run, when going through the worker playbook, or on the next run), the git checkout fails because the `key_file` variable is `None`.

When I turn on debugging, I can see why this is happening. The "upload github key" task succeeds the first time and returns its variable into the `github_key` register, where `github_key.dest` is the path to the deploy key. Any subsequent time through, however, Ansible is smart enough to see that the deploy key is already in place, so it doesn't do anything. But the return value in the register doesn't have a `dest` attribute, so `key_file` ends up empty. 

Here's an example of a successful run (of the 2 tasks in question): https://gist.github.com/vkurup/38d63b35a1edbc6b099c72846d156a44

Notice that the first task is `changed=true` and has a 'dest' attribute.

And a failed run:
https://gist.github.com/vkurup/fdd49ff4d3fdf6cae636803d00dff921

Notice that the first task is `changed=false` and does not have a 'dest' attribute.

The proper value IS in `github.path`, so I can fix things by changing the task to set `key_file` like so:

```yaml
- name: checkout latest project source
  git: repo={{ repo.url }}
       dest={{ source_dir }}
       version={{ repo.branch|default('master') }}
       force=yes
       ssh_opts="-o StrictHostKeyChecking=no"
       key_file={{ github_key.dest|default(github_key.path)|default(None) }}
  become_user: "{{ project_user }}"
  vars:
    ansible_ssh_pipelining: true
  when: not source_is_local
```

But I'm more wondering why this isn't affecting people on every deploy? I'm using Ansible 2.4.

**Comment by vkurup** [comment]
@jbradberry helped me identify that this is probably due to this bug in Ansible 2.3+: https://github.com/ansible/ansible/issues/25431

**Comment by vkurup** [comment]
Closed in v0.9.7 by #21
